internal/pkg/addon/addons.go

```diff
@@ -23,49 +23,53 @@ type workspaceReader interface {
 	ReadAddon(svcName, fileName string) ([]byte, error)
 }
 
-// Addons represents additional resources for a service.
+// Addons represents additional resources for a workload.
 type Addons struct {
-	svcName string
+	wlName string
+	wlType string
```

Comment:
Is the type really needed? I feel like an addon doesn't really care whether it's attached to a job or a service

Reply:
Should the type be established when `Template()` is called?

Reply:
Sorry I wasn't clear! Right now the `wlType` seems to be only used to make error messages clearer within the `addon` pkg, instead I'm thinking we should not add `wlType` as a field and the caller should decide whether they want to provide that information.

For example, the caller of `t, err := Template()` could do `fmt.Errorf("generate template for job: %w", err)`. Personally, I don't think the workload type is needed in these error messages since the user knows whether they ran `job package` or `svc package`

Reply:
Ah, gotcha. Thanks for the explanation! Taking out the workload type; I agree that the user should know which they mean. And I won't wrap `ErrAddonsDirNotExist` if I have `job_package` use `packageSvcOpts`.